### PR TITLE
Allow GitHub Staff members to be staff on Classroom for GitHub

### PR DESCRIPTION
**Pull request by tarebyte**
If the user is a staff member (the `site_admin` column) on GitHub then when they login they will
become staff on Classroom for GitHub.

Or if their uid is included in the NON_STAFF_GITHUB_ADMIN_IDS string
then they will also be updated to become an admin on sign in (for people like me :grin:)

/cc @johndbritton

### CODE REVIEW

**Comment by tarebyte** [comment]
In your .env file it would looking something like this

`NON_STAFF_GITHUB_ADMIN_IDS=8675309,42`

File: app/services/auth_hash.rb [hunk 1]
```diff
@@ -26,10 +27,16 @@ def token
...
+    raw_info[:site_admin]
+  end
+
   private
 
-  def info
-    user_hash.fetch('info', {})
+  def non_staff_github_admins_ids
+    return [] unless ENV['NON_STAFF_GITHUB_ADMIN_IDS'].present?
+    ENV['NON_STAFF_GITHUB_ADMIN_IDS'].split(',')
```
Comment (johndbritton) [inline app/services/auth_hash.rb]: You probably wants a `compact` here to make sure `nil` never ends up in that array. I'd also add a test case to make sure setting the environment variable to `,,1` does not include a nil in the resulting array.

File: spec/services/auth_hash_spec.rb [hunk 2]
```diff
@@ -1,8 +1,28 @@
...
+  let(:auth_hash) { AuthHash.new(OmniAuth.config.mock_auth[:github]) }
+
+  describe '#extract_user_info' do
+    it 'extracts the users information' do
+      expect(auth_hash.user_info[:token]).to eq('some-token')
+    end
+  end
+
+  describe '#non_staff_github_admins_ids' do
+    it 'returns an empty array if there the ENV is not set' do
```
Comment (johndbritton) [inline spec/services/auth_hash_spec.rb]: typo :smile:
